src/store/authSlice.js

```diff
@@ -0,0 +1,23 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+const initialState = {
+  userInfo: null,
```

Comment:
can spread the userInfo into different components linked username, email, first name instead of storing everything as a single object.